<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1089</width>
    <height>808</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>180</y>
      <width>301</width>
      <height>201</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QPushButton" name="Roberts">
       <property name="text">
        <string>Roberts operator</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="Prewitt">
       <property name="text">
        <string>Prewitt operator</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="Sobel">
       <property name="text">
        <string>Sobel operator</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_original_image">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>The Original Image</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_processed_image">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>380</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>The processed Image</string>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>180</y>
      <width>301</width>
      <height>201</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QPushButton" name="Gaussian">
       <property name="text">
        <string>Gaussian filter</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="Mean">
       <property name="text">
        <string>Mean filter(Kernel Size=3)</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="Median">
       <property name="text">
        <string>Median filter(Kernel Size=3)</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="loadPicture">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>80</y>
      <width>299</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Click to Load Your Picture</string>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>470</y>
      <width>551</width>
      <height>231</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QLabel" name="label_Kernel_Sigma">
       <property name="font">
        <font>
         <family>3ds</family>
         <pointsize>14</pointsize>
        </font>
       </property>
       <property name="text">
        <string> The Parameters :Kernel Size and Sigma are designed for Gaussian Filter</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_Kernel">
       <property name="text">
        <string>Kernal Size (Odd number from 3 to15)</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="Kernel_Size">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>7</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>2</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_SIgma">
       <property name="text">
        <string>Sigma(from 0.1 to 10.0)</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="Sigma">
       <property name="minimum">
        <number>10</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="min_of_Kernel__3">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>580</y>
      <width>72</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>3</string>
    </property>
   </widget>
   <widget class="QLabel" name="max_of_Kernel_15">
    <property name="geometry">
     <rect>
      <x>1050</x>
      <y>580</y>
      <width>72</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>15</string>
    </property>
   </widget>
   <widget class="QLabel" name="min_of_Sigma_1e1">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>680</y>
      <width>72</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>1.0</string>
    </property>
   </widget>
   <widget class="QLabel" name="max_of_Sigma_10">
    <property name="geometry">
     <rect>
      <x>1050</x>
      <y>680</y>
      <width>72</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>10.0</string>
    </property>
   </widget>
   <widget class="QLabel" name="original_image">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>401</width>
      <height>351</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="text">
     <string>The Original Image</string>
    </property>
   </widget>
   <widget class="QLabel" name="processed_image">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>410</y>
      <width>401</width>
      <height>341</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="text">
     <string>The Processed Image</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1089</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
